### ISSUE TITLE
Insufficient information on mock failure when argument is object

### ISSUE THREAD

**Issue by rickmugridge** [florinn/typemoq#56]
Consider the following test, which is the simplest example to show a problem with the error information:

       it(`With an object as a parameter, but the wrong one, we get insufficient information back from TypeMoq`, () => {
            let mock = TypeMoq.Mock.ofType(B, TypeMoq.MockBehavior.Strict);
            mock.setup(g => g.g({a:22})).returns(() => -222);
            expect(mock.object.g({a:11})).eql(222);
        });

The message is:
     MockException - no setup expression found for ('B.g([object Object])'
 
This works fine with simple values, of course.

There is an easy fix in this particular case. But when there are complex objects involved as arguments, it would be great to have displayed the full details of the objects concerned, both expected and actual. In my real case, I'm dealing with multiple arguments, all of which are complex objects and/or arrays.

**Comment by jezzsantos** [comment]
I am seeing this too.

Moq (C#) has been awesome at providing very detailed clues about what didn't work when verifying expectations. For example, in this code (that fails because of an incorrect expectation):

```
                this.addressResolver.Setup(pm => pm.ResolveStreetAddress(It.IsAny<String>()))
                    .Returns((Position)null);

                this.manager.CreateCar("adescription", "alicenseplate", "amake", "amodel", 2010));

                this.addressResolver.Verify(pm => pm.ResolveStreetAddress("avalidhomelocation"));
```

yields this kind of error detail, when the test fails due to failed expectation.

```
Moq.MockException: 
Expected invocation on the mock at least once, but was never performed: pm => pm.ResolveStreetAddress("avalidhomelocation")

Configured setups:
pm => pm.ResolveStreetAddress(It.IsAny<String>()), Times.Once

Performed invocations:
IAddressResolver.ResolveStreetAddress("aninvalidhomelocation")
<stacktrace following>
```

Can TypeMoq provide a similar output that lists: at least the expected value and the actual values that were returned like Moq does? Does not have to be exactly the same, but at least give enough clues as to where to focus the developer?

**Comment by florinn** [comment]
@rickmugridge Indeed `[object Object]` is not very informative. I've pushed a commit with some changes to display object args as json.

@jezzsantos Good idea. I've added to the failed verification message the list of the expected calls and their expected call counts (available only for the setups marked as `verifiable`) and the list of the actual calls.

Please install the development version and let me know how it goes.

**Comment by rickmugridge** [comment]
Many thanks @florinn ! I have been using the dev version for the last few days and the new display is excellent.

**Comment by florinn** [comment]
Thanks, Rick. I've just published a new version (1.5.0) that includes these changes.
